clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2279,25 +2280,65 @@ static bool CheckMaskedBuiltinArgs(Sema &S, Expr *MaskArg, Expr *PtrArg,
   if (!PtrTy->isPointerType() || PtrTy->getPointeeType()->isVectorType())
     return S.Diag(PtrArg->getExprLoc(), diag::err_vec_masked_load_store_ptr)
            << Pos << "scalar pointer";
+
+  QualType PointeeTy = PtrTy->getPointeeType();
+  if (PointeeTy.isVolatileQualified() || PointeeTy->isAtomicType() ||
+      (!AllowConst && PointeeTy.isConstQualified()) ||
+      (!AllowAS && PointeeTy.hasAddressSpace())) {
+    QualType Target =
+        S.Context.getPointerType(PointeeTy.getAtomicUnqualifiedType());
+    return S.Diag(PtrArg->getExprLoc(),
+                  diag::err_typecheck_convert_incompatible)
+           << PtrTy << Target << /*different qualifiers=*/5
+           << /*qualifier difference=*/0 << /*parameter mismatch=*/3 << 2
+           << PtrTy << Target;
+  }
+  return false;
+}
+
+static bool ConvertMaskedBuiltinArgs(Sema &S, CallExpr *TheCall,
+                                     unsigned PtrArg) {
+  bool TypeDependent = false;
+  for (unsigned Arg = 0, E = TheCall->getNumArgs(); Arg != E; ++Arg) {
+    ExprResult Converted =
+        Arg == PtrArg
+            ? S.DefaultFunctionArrayLvalueConversion(TheCall->getArg(Arg))
+            : S.DefaultLvalueConversion(TheCall->getArg(Arg));
```

Comment:
Just do FunctionArrayLvalueConversion in both cases. The difference is just that arrays and functions become pointers, and your other arguments can't be arrays, functions, or pointers, so it's not actually harmful to do the full set of conversions in all positions. If someone comes along and wants to clean up this code to consistently do conversions, I don't want them to get caught up thinking that there's an important difference they need to preserve here.